### PR TITLE
Improved repo-check test [repo-check-dev]

### PR DESCRIPTION
**Pull request by tzanio**
As @vsoch noticed in [#2981](https://github.com/mfem/mfem/pull/2981#issuecomment-1111644170), we don’t run certain tests (`repo-check`) from forks, which is confusing for new users. The [reason for that ](https://github.com/mfem/mfem/pull/2981#issuecomment-1113652600) is that these are the checks we want to run more frequently — on all pushes, for branches with or without a PR.

This PR fixes this by running the `repo-check` test on both pushes and PR updates, skipping duplicate runs (on pushes to local PRs). 

- This is demonstrated in [#2989](https://github.com/mfem/mfem/pull/2989) and [#2990](https://github.com/mfem/mfem/pull/2990).
- The advantage is that `repo-check` tests are run for PRs from forks.
- The disadvantage is that we now have “skipped” tests for local PRs. It looks like there is no way not to show them.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge|
 | --- | --- | --- | --- | --- | --- | --- |
| [#3001](https://github.com/mfem/mfem/pull/3001) | @tzanio | @v-dobrev | @pazner + @mlstowell + @v-dobrev | 5/11/22 | 5/13/22 | 5/16/22 | |

<details>
<summary>PR Checklist</summary>
    
- [ ] Code builds.
- [ ] Code passes `make style`.
- [ ] Update `CHANGELOG`:
    - [ ] Is this a new feature users need to be aware of? New or updated example or miniapp?
    - [ ] Does it make sense to create a new section in the `CHANGELOG` to group with other related features?
- [ ] Update `INSTALL`:
    - [ ] Had a new optional library been added? If so, what range of versions of this library are required? (*Make sure the external library is compatible with our BSD license, e.g. it is not licensed under GPL!*)
    - [ ] Have the version ranges for any required or optional libraries changed?
    - [ ] Does `make` or `cmake` have a new target?
    - [ ] Did the requirements or the installation process change? *(rare)*
- [ ] Update continuous integration server configurations if necessary (e.g. with new version requirements for each of MFEM's dependencies)
    - [ ] `.github`
    - [ ] `.appveyor.yml`
- [ ] Update `.gitignore`:
    - [ ] Check if `make distclean; git status` shows any files that were generated from the source by the project (not an IDE) but we don't want to track in the repository.
    - [ ] Add new patterns (just for the new files above) and re-run the above test.
- [ ] New examples:
    - [ ] All sample runs at the top of the example source file work.
    - [ ] Update `examples/makefile`:
      - [ ] Add the example code to the appropriate `SEQ_EXAMPLES` and `PAR_EXAMPLES` variables.
      - [ ] Add any files generated by it to the `clean` target.
      - [ ] Add the example binary and any files generated by it to the top-level `.gitignore` file.
    - [ ] Update `examples/CMakeLists.txt`:
      - [ ] Add the example code to the `ALL_EXE_SRCS` variable.
      - [ ] Make sure `THIS_TEST_OPTIONS` is set correctly for the new example.
   - [ ] List the new example in `doc/CodeDocumentation.dox`.
   - [ ] If new examples directory (e.g.`examples/pumi`), list it in `doc/CodeDocumentation.conf.in`
   - [ ] Companion pull request for documentation in [mfem/web](https://github.com/mfem/web) repo:
      - [ ] Update or add example-specific documentation, see e.g. the `src/examples.md`.
      - [ ] Add the description, labels and screenshots in `src/examples.md` and `src/img`.
      - [ ] In `examples.md`, list the example under the appropriate categories, add new categories if necessary.
      - [ ] Add a short description of the example in the "Extensive Examples" section of `features.md`.
- [ ] New miniapps:
   - [ ] All sample runs at the top of the miniapp source file work.
   - [ ] Update top-level `makefile` and `makefile` in corresponding miniapp directory.
   - [ ] Add the miniapp binary and any files generated by it to the top-level `.gitignore` file.
   - [ ] Update CMake build system:
      - [ ] Update the `CMakeLists.txt` file in the `miniapps` directory, if the new miniapp is in a new directory.
      - [ ] Add/update the `CMakeLists.txt` file in the new miniapp directory.
      - [ ] Consider adding a new test for the new miniapp.
   - [ ] List the new miniapp in `doc/CodeDocumentation.dox`
   - [ ] If new miniapps directory (e.g.`miniapps/nurbs`), add it to `MINIAPP_SUBDIRS` in the `makefile`.
   - [ ] If new miniapps directory (e.g.`miniapps/nurbs`), list it in `doc/CodeDocumentation.conf.in`
   - [ ] Companion pull request for documentation in [mfem/web](https://github.com/mfem/web) repo:
     - [ ] Update or add miniapp-specific documentation, see e.g. the `src/meshing.md` and `src/electromagnetics.md` files.
     - [ ] Add the description, labels and screenshots in `src/examples.md` and `src/img`.
     - [ ] The miniapps go at the end of the page, and are usually listed only under a specific "Application (PDE)" category.
     - [ ] Add a short description of the miniapp in the "Extensive Examples" section of `features.md`.
- [ ] New capability:
   - [ ] All new public, protected, and private classes, methods, data members, and functions have full Doxygen-style documentation in source comments. Documentation should include descriptions of member data, function arguments and return values, template parameters, and prerequisites for calling new functions.
   - [ ] Pointer arguments and return values must specify whether ownership is being transferred or lent with the call.
   - [ ] Any new functions should include descriptions of their intended use e.g. for internal use only, user-facing, etc., along with references to example code whenever possible/appropriate.
   - [ ] Consider adding new sample runs in existing examples to highlight the new capability.
   - [ ] Consider saving cool simulation pictures with the new capability in the Confluence gallery (LLNL only) or submitting them, via pull request, to the gallery section of the `mfem/web` repo.
   - [ ] If this is a major new feature, consider mentioning it in the short summary inside `README` *(rare)*.
   - [ ] List major new classes in `doc/CodeDocumentation.dox` *(rare)*.
- [ ] Update this checklist, if the new pull request affects it.
- [ ] Run `make unittest` to make sure all unit tests pass.
- [ ] Run the tests in `tests/scripts`.
- [ ] (LLNL only) After merging:
   - [ ] Update internal tests to include the new features.
</details>

### CODE REVIEW

**Comment by tzanio** [comment]
Some excerpts from the GH-Actions documentation for future reference (see [here](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#using-multiple-events) and [here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflows-in-forked-repositories)):

- If you specify multiple events, only one of those events needs to occur to trigger your workflow. If multiple triggering events for your workflow occur at the same time, multiple workflow runs will be triggered.
- For pull requests from a forked repository to the base repository, GitHub sends the pull_request, issue_comment, pull_request_review_comment, pull_request_review, and pull_request_target events to the base repository. No pull request events occur on the forked repository.
- When a first-time contributor submits a pull request to a public repository, a maintainer with write access may need to approve running workflows on the pull request.
- Workflows don't run in forked repositories by default. You must enable GitHub Actions in the Actions tab of the forked repository.

File: .github/workflows/repo-check.yml
Comment (v-dobrev) [inline .github/workflows/repo-check.yml:32]: It may be better to split long lines like this using the multi-line yaml syntax, e.g. something like this:
```yaml
     if: |
       (github.event_name == 'push' ||
        github.event.pull_request.head.repo.full_name != github.repository)
```
See https://github.community/t/how-to-write-multi-line-condition-in-if/128477.

Comment (tzanio) [reply .github/workflows/repo-check.yml]: done in https://github.com/mfem/mfem/pull/3001/commits/b30b091ae9f3a4f33be19ca85efa2d8351ca302a

**Review by v-dobrev — APPROVED** [review]

**Comment by v-dobrev** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Review by psocratis — APPROVED** [review]

**Review by pazner — APPROVED** [review]

**Comment by v-dobrev** [comment]
Merged in `next` for testing...

**Comment by v-dobrev** [comment]
Since we have enough approvals I merged the PR in `next` for testing -- since the changes are just in the GitHub actions, the testing in `next` is just a formality.

I'll wait a little longer before merging the PR in `master` in case any other reviewers want to take a look at this.

**Review by adrienbernede — APPROVED** [review]

**Comment by tzanio** [comment]
@psocratis, you can run the tests manually now on a branch without PR, see the `Run workflow` button in

- https://github.com/mfem/mfem/actions/workflows/repo-check.yml
- https://github.com/mfem/mfem/actions/workflows/builds-and-tests.yml
- https://github.com/mfem/mfem/actions/workflows/mfem-analysis.yml

**Pull request merged**
